src/intervals/arithmetic.jl

```diff
@@ -174,24 +170,62 @@ function /{T<:Real}(a::Interval{T}, b::Interval{T})
 
         end
     end
-    # a * inv(b)
-    # @round(S, min( a.lo/b.lo, a.lo/b.hi, a.hi/b.lo, a.hi/b.hi ),
-    #           max( a.lo/b.lo, a.lo/b.hi, a.hi/b.lo, a.hi/b.hi ) )
 end
 
 //(a::Interval, b::Interval) = a / b    # to deal with rationals
 
 
+## fma: fused multiply-add
+function fma(a::Interval, b::Interval, c::Interval)
+    T = promote_type(eltype(a), eltype(b), eltype(c))
+
+    (isempty(a) || isempty(b) || isempty(c)) && return emptyinterval(T)
+
+    if isentire(a)
+        b == zero(b) && return c
+        return entireinterval(T)
+    elseif isentire(b)
+        a == zero(a) && return c
+        return entireinterval(T)
+    end
+
+    lo = with_rounding(T, RoundDown) do
+        lo1 = fma(a.lo, b.lo, c.lo)
+        lo2 = fma(a.lo, b.hi, c.lo)
+        lo3 = fma(a.hi, b.lo, c.lo)
+        lo4 = fma(a.hi, b.hi, c.lo)
+        min(lo1, lo2, lo3, lo4)
+    end
+    hi = with_rounding(T, RoundUp) do
+        hi1 = fma(a.lo, b.lo, c.hi)
+        hi2 = fma(a.lo, b.hi, c.hi)
+        hi3 = fma(a.hi, b.lo, c.hi)
+        hi4 = fma(a.hi, b.hi, c.hi)
+        max(hi1, hi2, hi3, hi4)
+    end
+    Interval(lo, hi)
+end
+
+
 ## Scalar functions on intervals (no directed rounding used)
 
 function mag(a::Interval)
     isempty(a) && return convert(eltype(a), NaN)
-    max( abs(a.lo), abs(a.hi) )
+    T = eltype(a)
+    r1, r2 = with_rounding(T, RoundUp) do
```

Comment:
Not sure that rounding here is really necessary, since I guess `abs` just changes the sign, which should not involve rounding, and `max` also doesn't involve rounding. Still, I guess it doesn't hurt.